core/src/test/java/com/linecorp/armeria/server/DefaultUnhandledExceptionReporterTest.java

```diff
@@ -0,0 +1,123 @@
+    private static final long awaitIntervalMillis = 2000;
+    private static volatile boolean throwNonIgnorableException;
+
+    @BeforeEach
+    public void attachAppender() {
+        logAppender.start();
+        errorHandlerLogger.addAppender(logAppender);
+        throwNonIgnorableException = false;
+    }
+
+    @AfterEach
+    public void detachAppender() {
+        errorHandlerLogger.detachAppender(logAppender);
+        logAppender.list.clear();
+    }
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.route()
+              .get("/logging-service-subsequently-decorated")
+              .decorator(service -> (ctx, req) -> {
+                  if (throwNonIgnorableException) {
+                      throw new IllegalArgumentException("Non-ignorable exception");
+                  }
+                  return service.serve(ctx, req);
+              })
+              .decorator(LoggingService.newDecorator())
+              .build((ctx, req) -> {
+                  throw HttpStatusException.of(HttpStatus.BAD_REQUEST);
+              });
+            sb.route()
+              .get("/logging-service-previously-decorated")
+              .decorator(LoggingService.newDecorator())
+              .decorator(service -> (ctx, req) -> {
+                  if (throwNonIgnorableException) {
+                      throw new IllegalArgumentException("Non-ignorable exception");
+                  }
+                  return service.serve(ctx, req);
+              })
+              .build((ctx, req) -> {
+                  throw HttpStatusException.of(HttpStatus.BAD_REQUEST);
+              });
+            sb.unhandledExceptionsReportInterval(Duration.ofMillis(reportIntervalMillis));
+        }
+    };
+
+    @Test
+    void allExceptionShouldNotBeReportedWhenLoggingServiceIsSubsequentlyDecorated() throws Exception {
+        server.blockingWebClient().get("/logging-service-subsequently-decorated");
+        throwNonIgnorableException = true;
+        server.blockingWebClient().get("/logging-service-subsequently-decorated");
+        Thread.sleep(reportIntervalMillis + awaitIntervalMillis);
+        assertThat(logAppender.list).isEmpty();
+    }
+
+    @Test
+    void nonIgnorableExceptionShouldBeReportedWhenLoggingServiceIsPreviouslyDecorated() {
+        throwNonIgnorableException = true;
+        server.blockingWebClient().get("/logging-service-previously-decorated");
+        await().atMost(Duration.ofMillis(reportIntervalMillis + awaitIntervalMillis))
+               .untilAsserted(() -> assertThat(logAppender.list).isNotEmpty());
+
+        assertThat(logAppender.list
+                           .stream()
+                           .filter(event -> event.getFormattedMessage().contains(
+                                   "Observed 1 exceptions that didn't reach a LoggingService"))
```

Comment:
How about updating the original message like this:
```suggestion
                                   "Observed 1 exception(s) that didn't reach a LoggingService"))
```

Reply:
Sure! 👍